includes/apple-exporter/components/class-recipe.php

```diff
@@ -0,0 +1,1206 @@
+				'#recipe_details_size#'                  => (int) $theme->get_value( 'recipe_details_size' ),
+				'#recipe_details_line_height#'           => (int) $theme->get_value( 'recipe_details_line_height' ),
+				'#recipe_details_tracking#'              => (int) $theme->get_value( 'recipe_details_tracking' ) / 100,
+				'#recipe_details_color#'                 => $theme->get_value( 'recipe_details_color' ),
+				'#recipe_details_color_dark#'            => $theme->get_value( 'recipe_details_color_dark' ),
+				'#recipe_details_link_color#'            => $theme->get_value( 'recipe_details_link_color' ),
+				'#recipe_details_link_color_dark#'       => $theme->get_value( 'recipe_details_link_color_dark' ),
+				'#recipe_details_background_color#'      => $theme->get_value( 'recipe_details_background_color' ),
+				'#recipe_details_background_color_dark#' => $theme->get_value( 'recipe_details_background_color_dark' ),
+			],
+			'textStyle',
+		);
+	}
+
+	/**
+	 * Set the JSON for the component.
+	 *
+	 * @param string $spec_name The spec to use for defining the JSON.
+	 * @param array  $values    Values to substitute for placeholders in the spec.
+	 */
+	protected function register_json( $spec_name, $values = [] ) {
+		if ( 'json' === $spec_name ) {
+			$values = $this->prepare_tokens( $values );
+		}
+
+		// Go through normal token replacement.
+		parent::register_json( $spec_name, $values );
+
+		if ( 'json' === $spec_name ) {
+			if ( is_array( $this->json ) ) {
+				$spec = $this->get_spec( $spec_name );
+
+				if ( $spec instanceof Component_Spec ) {
+					// Update the JSON to remove objects with recipe tokens that weren't replaced.
+					$this->json = $this->without_objects_containing_tokens( $this->json, $spec );
+				}
+			}
+		}
+	}
+
+	/**
+	 * Find JSON-LD Recipe items for the given recipe HTML.
+	 *
+	 * @param string $recipe The recipe HTML found in the matching node for this component instance.
+	 * @return array|null The matching schema, or null if none found.
+	 */
+	private function matching_schema( $recipe ): ?array {
+		// First, check the post content for a JSON-LD Recipe.
+		$post = get_post( $this->workspace->content_id );
+
+		/** This filter is documented in admin/apple-actions/index/class-export.php */
+		$post_content = apply_filters( 'apple_news_exporter_content_pre', $post->post_content, $post->ID );
+
+		/** This filter is documented in wp-includes/post-template.php */
+		$post_content = apply_filters( 'the_content', $post_content ); // phpcs:ignore WordPress.NamingConventions.PrefixAllGlobals.NonPrefixedHooknameFound
+
+		$items = self::recipe_items( $post_content, 'body' );
+
+		foreach ( $items as $item ) {
+			if ( self::is_recipe_item_for_recipe_html( $item, $recipe ) ) {
+				return $item;
+			}
+		}
+
+		// Second, try to find schema in the HEAD of the post on the frontend.
+		// Could check BODY as well, but the post content is already being scanned. That might be enough.
```

Comment:
Any update needed to this comment after testing in development?

Reply:
This is the "serious" revision of the comment 😆 But I wanted to leave it because I don't really know whether it will be confirmed when actual sites start to use the feature and leave feedback about whether the schema is being picked up.